pytools/tag.py

```diff
@@ -94,6 +95,28 @@ def from_class(cls, argcls: Any) -> "DottedName":
 tag_dataclass = dataclass(init=True, eq=True, frozen=True, repr=True)
 
 
+@tag_dataclass
+class Taggable:
+    """
+    Parent class for objects with a `tags` attribute.
+
+    .. attribute:: tags
+
+        An instance of or a data structure containing instances of :class:`Tag`
+    .. automethod:: __repr__
+    """
+    def __init__(self, tags=None):
+        self._tags = tags
+
+    @property
+    def tags(self):
+        return self._tags
+
+    @tags.setter
+    def tags(self, tags):
+        self._tags = tags
```

Comment:
I wasn't imagining that this would support in-place modification. Instead, add a `tagged()` method that returns a copy that has the tag added. (Assume that you can call `self.copy(tags=<NEW VALUE>)` and document this assumption.

Reply:
Okay, added that.